src/js/control-bar/volume-control/check-volume-support.js

```diff
@@ -1,3 +1,36 @@
+/**
+ * A helper function used by {@link checkVolumeSupport} to hide volume/mute toggle
+ * button when the actions by them are unsopported in the current tech.
+ *
+ * @param {Component} self
+ *        The component that should be hidden if volume is unsupported
+ *
+ * @param {Player} player
+ *        A reference to the player
+ *
+ * @private
+ */
+const checkVolumeSupportHelper = function(self, player) {
+  const isMuteToggle = self.name_ === 'MuteToggle';
+
+  // hide volume panel if neither mute or volume control
+  // is supported by the current tech
+  if (player.tech_ && !player.tech_.featuresVolumeControl && !player.tech_.featuresMuteControl) {
+    player.getChild('controlBar').getChild('volumePanel').addClass('vjs-hidden');
+  // hide volume slider if volume control
+  // is not supported by the current tech
+  } else if (player.tech_ && !player.tech_.featuresVolumeControl && !isMuteToggle) {
+    self.addClass('vjs-hidden');
+    player.getChild('controlBar').getChild('volumePanel').addClass('vjs-mute-supported');
```

Comment:
is just removing `vjs-hidden` from the mute-toggle not enough?
Maybe it'll be easier with my suggestion of splitting the functionality into two methods instead of trying to do it in one method.

Reply:
Volume panel expands when active or moused over. So if volume control isn't shown, you get empty space between mute toggle and progress bar. Additional class is used to prevent this behaviour when only mute toggle is present.